hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -339,159 +227,176 @@ public void noMoreTransactions() {
 	@Override
 	public synchronized ServicesState copy() {
 		setImmutable(true);
-		final var mutableUniqTokenAssocsIfInit =
-				(uniqueTokenAssociations == null) ? null : uniqueTokenAssociations.copy();
-		final var mutableOwnerAssocsIfInit =
-				(uniqueOwnershipAssociations == null) ? null : uniqueOwnershipAssociations.copy();
-		final var mutableOwnerTreasuryAssocsIfInit =
-				(uniqueTreasuryOwnedAssociations == null) ? null : uniqueTreasuryOwnedAssociations.copy();
-		return new ServicesState(ctx, nodeId, List.of(
-				addressBook().copy(),
-				networkCtx().copy(),
-				topics().copy(),
-				storage().copy(),
-				accounts().copy(),
-				tokens().copy(),
-				tokenAssociations().copy(),
-				diskFs().copy(),
-				scheduleTxs().copy(),
-				runningHashLeaf().copy(),
-				uniqueTokens().copy()
-		), mutableUniqTokenAssocsIfInit, mutableOwnerAssocsIfInit, mutableOwnerTreasuryAssocsIfInit, this);
-	}
 
+		final var that = new ServicesState(this);
+		if (metadata != null) {
+			metadata.getCtx().update(that);
+		}
+
+		return that;
+	}
 
 	/* --- Archivable --- */
 	@Override
 	public void archive() {
-		releaseFcotmrIfNonNull();
+		if (metadata != null) {
+			metadata.archive();
+		}
 	}
```

Comment:
This method is also required to archive any `FCMap`s that are descendants. Although archive is currently disabled in the SDK version you are using, when archive is re-enabled (probably in 19) it will not work properly unless the `FCMap`s are properly handled.

Reply:
Added comment noting this.